flink-connectors/flink-connector-kafka-0.9/src/main/java/org/apache/flink/streaming/connectors/kafka/internal/KafkaConsumerThread.java

```diff
@@ -80,6 +83,9 @@
 	/** The queue of unassigned partitions that we need to assign to the Kafka consumer. */
 	private final ClosableBlockingQueue<KafkaTopicPartitionState<TopicPartition>> unassignedPartitionsQueue;
 
+	/** The list of partitions to be removed from kafka consumer. */
+	private final Set<TopicPartition> partitionsToBeRemoved;
```

Comment:
Would it actually make more sense that we have a queue for this? Like how we are handling unassigned new partitions via the `unassignedPartitionsQueue`. The fact that this is a set means that we will need to eventually remove entries from it anyways.

Reply:
From my understanding, for unassigned partitions we can use a Queue because it does not matter which consumer will take the new partitions.
But we can not use a Queue for partitions to be removed because we only can remove the partitions from the consumer that is actually subscribed to that partition.
Does that make sense ?